c7n/resources/ec2.py

```diff
@@ -295,6 +297,68 @@ def __call__(self, i):
         return self.operator(map(self.match, volumes))
 
 
+@filters.register('stop-protected')
+class DisableApiStop(Filter):
+    """EC2 instances with ``disableApiStop`` attribute set
+
+    Filters EC2 instances with ``disableApiStop`` attribute set to true.
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: stop-protection-enabled
+            resource: ec2
+            filters:
+              - type: stop-protected
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: stop-protection-NOT-enabled
+            resource: ec2
+            filters:
+              - not:
+                - type: stop-protected
+    """
+
+    schema = type_schema('stop-protected')
+    permissions = ('ec2:DescribeInstanceAttribute',)
+
+    def get_permissions(self) -> List[str]:
+        perms = list(self.permissions)
+        perms.extend(self.manager.get_permissions())
+        return perms
+
+    def process(self, resources: List[dict], event=None) -> List[dict]:
+        client = utils.local_session(
+            self.manager.session_factory).client('ec2')
+        return [r for r in resources
+                if self._is_stop_protection_enabled(client, r)]
+
+    def _is_stop_protection_enabled(self, client, instance: dict) -> bool:
+        attr_val = self.manager.retry(
+            client.describe_instance_attribute,
+            Attribute='disableApiStop',
+            InstanceId=instance['InstanceId']
+        )
+        return attr_val['DisableApiStop']['Value']
+
+    def validate(self) -> None:
+        botocore_min_version = '1.26.7'
+
+        from pkg_resources import parse_version
```

Comment:
hmm.. I do wonder about pkg_resources vs other tools, there's some upstream work on trying to deprecate some of things which I'm not fully on top of.  

the pkg resources doc https://setuptools.pypa.io/en/latest/pkg_resources.html
basically says don't use it vs import lib.metadata in the standard lib. our usage here is more on the version check though which isn't in import lib / sodlib. distutils in stdlib has a version class that we could use, but that's also supposed to go extinct/deprecated.

Reply:
it looks like we already use distutils.LooseVersion in c7n in a few places, so we can use that in place of pkg_resources direct usage here, and we'll be able to handle its eventual deprecation along with the rest of our uses.

Reply:
I have also found the following ways to compare library versions (no custom parsing approach):
- packaging.version.parse
- pkg_resources.parse_version

It seems that the `packaging` library is the most suitable for that case (don't see _deprecated_ warnings). But there is no this library on AWS Lambda (Yet I only tested it on default `python3.8` lambda; not sure if C7n prepared lambda has additional libraries):
```json
{
  "errorMessage": "Unable to import module 'index': No module named 'packaging'",
  "errorType": "Runtime.ImportModuleError",
  "stackTrace": []
}
```
If you think it is worth checking `packaging` on C7n prepared lambda (e.g., c7n depends on it so that it will be available on lambda after adding c7n package), let me know and I'll test it.

So I used `pkg_resources` instead.

Just tested `distutils.version` (missed it before). It is available on `python3.8` lambda function.

Reply:
Refactored. `distutils.version.LooseVersion` is used now.